book/src/simple_application/drawing.md

```diff
@@ -0,0 +1,172 @@
+> yet.
+
+Firstly, we need to revisit the `State.onStart` trait method.
+
+```rust
+fn on_start(&mut self, 
+            world: &mut World, 
+            assets: &mut AssetManager, 
+            pipe: &mut Pipeline)
+```
+Let's talk about the parameters here. The first one - `&mut World` is the
+global object containing all entities, components, and systems. We can use it to
+add *entities* to the world. These entities can have an arbitrary set of
+*component* information associated with them, The entity is the thing, and the
+components are the bits of data attached to it (or more accurately, the types
+of information). *Systems* then drive the data in the entities for the
+components they are interested in/responsible for. I find it a bit confusing,
+so let's jump into pong to see if a real world example helps.
+
+When we want to add an entity, we call `world.create_now()` which returns a
+[builder pattern][builder_pattern] object. We can then call `with` on this
+object to add component information, and then when we have added everything we
+want to, we call `build` to finalise it into an entity in our world.
+Here is a full example for our pong ball.
+
+```rust
+world.create_now()
+    .with(square.clone())
+    .with(ball)
+    .with(LocalTransform::default())
+    .with(Transform::default())
+    .build();
+```
+Some of the components we can see here are built into amethyst to control how
+things are drawn, and we can add an unlimited number of our own components.
+We'll see how some of the built-in components work later.
+
+# Creating a ball entity
+
+We're going to spend the rest of this stage learning how to draw to the screen.
+We want to create an entity, and attach the required component information to
+it so amethyst can draw it.
+
+Let's look at the second parameter to `on_start`, a `&mut AssetManager`. The
+asset manager is responsible for holding rendering information, including
+meshes and textures. We're going to tell it to make space for our meshes and
+textures, then to store some very simple ones that we need to draw our ball. We
+will use a flat color for our texture, so there will be no need to load any
+image files, and we will use a simple square for our ball, containing only 2
+triangles.
+
+## Adding a renderable
+
+```rust
+
+// Generate a square mesh
+assets.register_asset::<Mesh>();
+assets.register_asset::<Texture>();
+assets.load_asset_from_data::<Texture, [f32; 4]>("white", [1.0, 1.0, 1.0, 1.0]);
+let square_verts = gen_rectangle(1.0, 1.0);
+assets.load_asset_from_data::<Mesh, Vec<VertexPosNormal>>("square", square_verts);
+let square = assets.create_renderable("square", "white", "white", "white", 1.0).unwrap();
+```
+
+The stages are
+ 1. Register asset types, so we can store assets of that type
+ 2. Load a single pixel texture into our asset manager. The texture is 4 `f32`s
+    wide, one for each of rgba. It is labelled with the string "white".
```

Comment:
maybe explain what the label is here. Does it have to be unique, is it just a description, when it is used?